Azure integration

aluminiumgeek opened the pull request:
Adds provider modules needed for integration with AKS.

Including:
- cluster initialization changes
- updated docs
- azuresql
- sql connection string updates to be compatible with azuresql format

aluminiumgeek committed:
Azure integration: initial docs and cluster initialization changes

aluminiumgeek committed:
- Azure DB connection settings
- Refactor DB connection settings to use kwargs instead of generating connection string
- Azure cluster settings

aluminiumgeek committed:
Add docs about DNS and routing

aluminiumgeek committed:
[WIP] Deis instance manager, restore later

aluminiumgeek committed:
Add missed azuresql db provider

aluminiumgeek committed:
credentials is a tuple in get_postgres_admin_credentials

aluminiumgeek committed:
Revert previous change

aluminiumgeek committed:
Remove unused functions

aluminiumgeek committed:
Add Azure storage module

akariv commented on ckan_cloud_operator/deis_ckan/envvars.py:
```diff
@@ -57,10 +57,10 @@ def _update(self):
-            CKAN___BEAKER__SESSION__URL=f"postgresql://{db_name}:{db_password}@{postgres_host}:{postgres_port}/{db_name}",
-            CKAN__DATASTORE__READ_URL=f"postgresql://{datastore_ro_user}:{datastore_ro_password}@{postgres_host}:{postgres_port}/{datastore_name}",
-            CKAN__DATASTORE__WRITE_URL=f"postgresql://{datastore_name}:{datastore_password}@{postgres_host}:{postgres_port}/{datastore_name}",
+            CKAN_SQLALCHEMY_URL=f"postgresql://{db_name}@{postgres_host}:{db_password}@{postgres_host}:{postgres_port}/{db_name}",
```
Why this change? it makes the urls invalid (and how are they related to azure?)

akariv replied:
@aluminiumgeek

aluminiumgeek replied:
Azure SQL instances support usernames only is this format ("username@host"). This format in URI works with patched version of CKAN for Gates.

I'll revert this change anyway, as we currently use helm based CKAN instances instead of deis.

akariv commented on ckan_cloud_operator/deis_ckan/instance.py:
```diff
@@ -312,7 +312,7 @@ def update(self, wait_ready=False, skip_solr=False, skip_deployment=False):
             logs.warning('Setting datastore permissions failed, continuing anyway')
         # Create/Update uptime monitoring after everything else is ready
-        DeisCkanInstanceUptime(self).update(envvars.site_url)
+        #DeisCkanInstanceUptime(self).update(envvars.site_url)
```
why?

zelima replied:
Should we just delete it if not needed?

akariv commented on ckan_cloud_operator/labels/manager.py:
```diff
@@ -11,6 +11,7 @@ def initialize(log_kwargs=None):
 
 def get_label_prefix(short=False):
     """Returns a global label prefix which should be used to namespace operator objects"""
+    return 'ckan-cloud'
```
looks like some debug code leftover (?)

aluminiumgeek replied:
right

akariv commented on ckan_cloud_operator/providers/ckan/db/migration.py:
```diff
@@ -438,7 +438,8 @@ def _create_base_dbs_and_roles(migration_name, db_name, datastore_name, recreate
             yield {'step': 'created-datastore-ro-role', 'msg': f'Created Datastore read-only user: {datastore_ro_name} ({db_prefix})'}
     yield {'step': 'created-base-dbs-and-roles', f'msg': f'Created base dbs and roles: {db_name}, {datastore_name}, {datastore_ro_name} ({db_prefix})'}
-    yield from _update_db_proxy(db_name, datastore_name, datastore_ro_name, db_password, datastore_password, datastore_ro_password, db_prefix)
+    if cluster_manager.get_provider_id() != 'azure':
```
why?

aluminiumgeek replied:
DB proxy module for azure wasn't ready. I pushed it.

akariv commented on ckan_cloud_operator/providers/cluster/azure/manager.py:
```diff
@@ -0,0 +1,130 @@
+#### standard provider code ####
+
+import subprocess
```
move to imports section below

akariv commented on ckan_cloud_operator/providers/cluster/azure/manager.py:
```diff
@@ -0,0 +1,130 @@
+    return binascii.hexlify(os.urandom(l)).decode()
+
+
+def check_output(cmd):
```
better rename to something more descriptive, e.g. `az_check_output`

akariv commented on ckan_cloud_operator/providers/db/azuresql/cli.py:
```diff
@@ -0,0 +1,19 @@
+
+
+@click.group('azuresql')
+def rds():
```
is rds an azure term?

akariv commented on ckan_cloud_operator/providers/db/azuresql/manager.py:
```diff
@@ -0,0 +1,85 @@
+
+from ckan_cloud_operator import logs
+from ckan_cloud_operator.providers.cluster import manager as cluster_manager
+from ckan_cloud_operator.drivers.gcloud import driver as gcloud_driver
```
why this import?

akariv commented on ckan_cloud_operator/providers/db/azuresql/manager.py:
```diff
@@ -0,0 +1,85 @@
+            _config_set(key, default_value, **_get_config_credentials_kwargs(db_prefix))
+        elif not config.get(key):
+            raise Exception(f'missing key: {key}')
+    #from ckan_cloud_operator.providers.db.proxy.azuresql import manager as proxy_azuresql_manager
```
remove commented out code

aluminiumgeek replied:
Uncommented it, as azuresql proxy module is ready

akariv commented on ckan_cloud_operator/providers/storage/azure/manager.py:
```diff
@@ -0,0 +1,106 @@
+from .constants import PROVIDER_ID
+
+
+def initialize(*args, **kwargs):
```
please provide explicit parameters to this function

zelima commented on ckan_cloud_operator/deis_ckan/envvars.py:
```diff
@@ -57,10 +57,10 @@ def _update(self):
+            CKAN_SQLALCHEMY_URL=f"postgresql://{db_name}@{postgres_host}:{db_password}@{postgres_host}:{postgres_port}/{db_name}",
+            CKAN___BEAKER__SESSION__URL=f"postgresql://{db_name}@{postgres_host}:{db_password}@{postgres_host}:{postgres_port}/{db_name}",
+            CKAN__DATASTORE__READ_URL=f"postgresql://{datastore_ro_user}@{postgres_host}:{datastore_ro_password}@{postgres_host}:{postgres_port}/{datastore_name}",
+            CKAN__DATASTORE__WRITE_URL=f"postgresql://{datastore_name}@{postgres_host}:{datastore_password}@{postgres_host}:{postgres_port}/{datastore_name}",
```
Could you please explain this? Why do we need `postgres_host` before username as well?

zelima replied:
Also, I remember I had a problem with connecting to azure database with URIs like this as username include `@` in it. Eg `psql postgresql://irakli.mchedlishvli@azure.com:passowrd@host:5000` was erroring cause it's getting confused with two `@`s in URL. Have we tested CKAN is fine with that?

zelima commented on ckan_cloud_operator/deis_ckan/instance.py:
```diff
@@ -338,7 +338,7 @@ def delete(self, force=False, wait_deleted=False):
                 'envvars-secret': lambda: kubectl.check_call(f'delete --ignore-not-found secret/{self.id}-envvars'),
                 'routes': lambda: routers_manager.delete_routes(deis_instance_id=self.id),
-                'uptime-monitoring': lambda: DeisCkanInstanceUptime(self).delete(self.id)
+                #'uptime-monitoring': lambda: DeisCkanInstanceUptime(self).delete(self.id)
```
Same here, do we actually need this comment?

zelima commented on ckan_cloud_operator/labels/manager.py:
```diff
@@ -11,6 +11,7 @@ def initialize(log_kwargs=None):
 
 def get_label_prefix(short=False):
     """Returns a global label prefix which should be used to namespace operator objects"""
+    return 'ckan-cloud'
```
I assume this is hardcoded for development purposes... (Or we don't need another `return` coming after)

zelima commented on ckan_cloud_operator/providers/ckan/db/migration.py:
```diff
@@ -438,7 +438,8 @@ def _create_base_dbs_and_roles(migration_name, db_name, datastore_name, recreate
     yield {'step': 'created-base-dbs-and-roles', f'msg': f'Created base dbs and roles: {db_name}, {datastore_name}, {datastore_ro_name} ({db_prefix})'}
-    yield from _update_db_proxy(db_name, datastore_name, datastore_ro_name, db_password, datastore_password, datastore_ro_password, db_prefix)
+    if cluster_manager.get_provider_id() != 'azure':
+        yield from _update_db_proxy(db_name, datastore_name, datastore_ro_name, db_password, datastore_password, datastore_ro_password, db_prefix)
```
Why not doing this for azure?

zelima commented on ckan_cloud_operator/providers/cluster/azure/manager.py:
```diff
@@ -0,0 +1,130 @@
+
+def initialize(interactive=False):
+    _set_provider()
+    if interactive:
+        print('\nEnter the Resource Group name\n')
+        _config_interactive_set({'azure-rg': None}, is_secret=False)
+        print('\nEnter the location of the Kubernets cluster is hosted on [westus2]\n')
+        _config_interactive_set({'azure-default-location': None}, is_secret=False)
+        print('\nEnter the name of your cluster\n')
+        _config_interactive_set({'azure-cluster-name': None}, is_secret=False)
```
What if not interactive? Will it just skip and do nothing? Should not we at least say smth like "please set --interactive flag" or similar?

akariv replied:
providers_manager.config_interactive_set has an 'interactive' parameter, better use it instead - that is, remove the if and pass interactive to every call of `_config_interactive_set`

zelima commented on ckan_cloud_operator/providers/db/azuresql/manager.py:
```diff
@@ -0,0 +1,85 @@
+        elif not config.get(key):
+            raise Exception(f'missing key: {key}')
+    #from ckan_cloud_operator.providers.db.proxy.azuresql import manager as proxy_azuresql_manager
+    #proxy_azuresql_manager.initialize(db_prefix=db_prefix)
```
Redundant comments?

zelima commented on docs/PRODUCTION-AZURE-CLUSTER.md:
````diff
@@ -0,0 +1,120 @@
+After that create `uptime-statuscake-api` secrets with keys "user", "key", "group" populated from StatusCake account:
+```
+kubectl -n ckan-cloud create secret generic uptime-statuscake-api --from-literal=user=<user> --from-literal=key=<key> --from-literal=group=<group>
+```
````
Why not this is the part of `ckan-cloud-operator cluster initialize --interactive`?

zelima replied:
Also, think statuscake stuff should be optional... What if I don't want to monitor uptime at all? Or want a different providers?

aluminiumgeek replied:
I removed info about it, because it's related only to deis instances.

zelima commented on docs/PRODUCTION-AZURE-CLUSTER.md:
```diff
@@ -0,0 +1,120 @@
+While interactive initialization:
+- Set `enable-deis-ckan: y`
+- If environment is production, set `env-id` to `p` on "routers" step.
+- On "solr" step of interactive initialization choose `self-hosted: y`
```
Just wondering, will this work if I set to `n` instead? Or I have to agree if I need to continue?

aluminiumgeek replied:
It will ask for http endpoint of an external solr instance.

zelima commented on docs/PRODUCTION-AZURE-CLUSTER.md:
````diff
@@ -0,0 +1,120 @@
+Give the service account permission to change cluster roles:
+```
+kubectl create clusterrolebinding default-sa-binding --clusterrole=cluster-admin --user=<service account email>
+```
````
Again kubectl command. Think we should wrap it with `ckan-cloud-opeartor`, no?

aluminiumgeek replied:
It's related only to GKE clusters, removed.

zelima reviewed (commented):
Great work! Mostly have some questions and suggestions, would be great if you could answer

akariv commented on docs/PRODUCTION-AZURE-CLUSTER.md:
```diff
@@ -0,0 +1,120 @@
+
+Download the file locally.
+
+## Create multi-user storage class
```
This also needs to be done in CCO initialisation.
See the AWS code (once you've rebased).

aluminiumgeek committed:
Add azuresql proxy module

aluminiumgeek committed:
Remove debugging stuff; Enable azuresql proxy initialization

aluminiumgeek committed:
Use `interactive` variable

akariv commented on ckan_cloud_operator/providers/cluster/gcloud/manager.py:
```diff
@@ -42,6 +42,8 @@ def initialize(interactive=False):
         _config_interactive_set({'cluster-name': None})
         print('\nEnter the google project ID\n')
         _config_interactive_set({'project-id': None})
+    else:
```
why this log message? you're not really skipping anything really...

akariv commented on ckan_cloud_operator/providers/db/proxy/azuresql/cli.py:
```diff
@@ -0,0 +1,35 @@
+
+@click.group()
+def azuresql():
+    """Manage the RDS centralized db proxy"""
```
another copy paste :)

akariv commented on ckan_cloud_operator/providers/db/proxy/azuresql/manager.py:
```diff
@@ -0,0 +1,150 @@
+        {
+            'replicas': 1,
+            'revisionHistoryLimit': 10,
+            'strategy': {'type': 'RollingUpdate', },
```
this won't work for k8s 1.16

aluminiumgeek replied:
I don't understand what will not work. Could you explain, please?

akariv replied:
afaik on k8s 1.16 it won't work without a `selector` entry.

aluminiumgeek replied:
It has a selector entry now.

akariv commented on ckan_cloud_operator/providers/storage/azure/constants.py:
```diff
@@ -0,0 +1 @@
+PROVIDER_ID = 'azure'
```
use a more accurate provider id ('azure' is already in use)

aluminiumgeek replied:
I'll rename it to `abs` (for Azure Blob Storage)

akariv replied:
please resolve only when done...

aluminiumgeek replied:
done

akariv reviewed (changes requested):
Some unanswered questions still...
+ some modifications in the deis code which shouldn't be

aluminiumgeek committed:
Merge branch 'master' into azure-integration

aluminiumgeek committed:
Python 3.6 compatibility

aluminiumgeek committed:
Use None value for default-root-domain in test

aluminiumgeek committed:
Add selector to spec

aluminiumgeek committed:
Move selector spec outside template

aluminiumgeek committed:
Get rid of magic number: check for sc an zk pod names

akariv commented on tests/routers/test_traefik.py:
```diff
@@ -36,7 +36,7 @@ def test_create_with_cloudflare_without_credentials(self, update_flag, get_provi
                 'name': 'datapusher',
             },
             'spec': {
-                'default-root-domain': 'default',
```
If the purpose of this test is to check that we assert on no credentials, then this fix won't be testing that.

aluminiumgeek replied:
Got it

akariv commented on ckan_cloud_operator/providers/db/proxy/azuresql/manager.py:
```diff
@@ -102,6 +102,11 @@ def _apply_deployment(db_prefix=None):
             'strategy': {'type': 'RollingUpdate', },
+            'selector': {
+                'matchLabels': {
+                    'app': 'provider-db-proxy-azuresql'
```
we have a function to generate the labels - why not use it?
(i.e. `_get_resource_labels(for_deployment=True, suffix=db_prefix or ''),`)

aluminiumgeek replied:
done

akariv replied:
where?

OriHoch commented:
:+1:

aluminiumgeek committed:
Revert change with usernames in "username@dbhost" format

aluminiumgeek committed:
Remove logs statement

aluminiumgeek committed:
- Rename PROVIDER_ID for azure storage
- Create cca-ckan stoarge class on azure cluster init
- Update docstring
- Remove info about creating storage class from documentation as it's automated

aluminiumgeek committed:
- Update docs
- Update fix for test_create_with_cloudflare_without_credentials test

akariv commented:
@aluminiumgeek 
please add a comment 'done' or 'fixed' when a CR issue was fixed.
it's up to the reviewer to resolve the issue or not.

akariv commented on ckan_cloud_operator/providers/solr/solrcloud/manager.py:
```diff
@@ -94,13 +94,11 @@ def initialize(interactive=False, dry_run=False):
         pods = kubectl.get('pods')
         running = len([x for x in pods['items']
-                       if x['status']['phase'] == 'Running'])
+                       if x['status']['phase'] == 'Running' and ('solrcloud-zk' in x['metadata']['name'] or 'solrcloud-sc' in x['metadata']['name'])])
```
see my other comment on checking the ['labels']['app'] instead.

aluminiumgeek replied:
done

akariv commented on ckan_cloud_operator/providers/db/proxy/azuresql/constants.py:
```diff
@@ -0,0 +1 @@
+PROVIDER_ID='azuresql'
```
the proxy can't be the same provider id as the main db provider

aluminiumgeek replied:
`gcloudsql` and `rds` db providers has the same provider id as their proxies

akariv replied:
I guess that's also a potential bug, as they might overwrite one another's configuration... unless there's a hidden reason for this.

aluminiumgeek replied:
done

akariv commented on ckan_cloud_operator/providers/cluster/aws/manager.py:
```diff
@@ -41,6 +41,8 @@ def initialize(interactive=False):
         _config_interactive_set({'aws-default-region': None}, is_secret=True)
         print('\nEnter the name of your Amazon EKS cluster\n')
         _config_interactive_set({'eks-cluster-name': None}, is_secret=True)
+    else:
```
see comment below... about this block

aluminiumgeek replied:
done

aluminiumgeek committed:
Remove logs statement; get sc and zk deployment names from _get_resource_labels()

akariv commented on ckan_cloud_operator/providers/solr/solrcloud/manager.py:
```diff
@@ -97,8 +97,10 @@ def initialize(interactive=False, dry_run=False):
+        zk_app_name = _get_resource_labels(for_deployment=True, suffix='zk')['app']
         running = len([x for x in pods['items']
-                       if x['status']['phase'] == 'Running' and ('solrcloud-zk' in x['metadata']['name'] or 'solrcloud-sc' in x['metadata']['name'])])
+                       if x['status']['phase'] == 'Running' and (zk_app_name in x['metadata']['name'] or sc_app_name in x['metadata']['name'])])
```
please use x['metadata']['labels']['app']

aluminiumgeek replied:
done

akariv commented on ckan_cloud_operator/logs.py:
```diff
@@ -74,7 +74,7 @@ def log_subprocess_output(stdout, stderr):
     completed = subprocess.run(
         command, input=input, 
-        shell=True, check=True, capture_output=True
+        shell=True, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
```
why?

aluminiumgeek replied:
Compatibility with Python 3.6 (`capture_output` introduced in 3.7 and does [the same](https://github.com/python/cpython/blob/3.7/Lib/subprocess.py#L481))

akariv commented on ckan_cloud_operator/providers/ckan/db/migration.py:
```diff
@@ -81,12 +81,12 @@ def delete(name, delete_dbs=False):
     migration = crds_manager.get(CRD_SINGULAR, name=name, required=False) or {}
```
why is this file being modified?

aluminiumgeek replied:
Azure SQL requires username in `username@host` format. Connection string with two `@` is not supported in psycopg2, so we pass connection parameters as kwargs.

akariv replied:
The question is what is db_migration being used for in the context of Azure?

aluminiumgeek replied:
It has been used when I set it up with deis instances. It's not used anymore. Reverting it.

akariv commented on ckan_cloud_operator/providers/db/proxy/azuresql/manager.py:
```diff
@@ -0,0 +1,156 @@
+            'strategy': {'type': 'RollingUpdate', },
+            'selector': {
+                'matchLabels': {
+                    'app': app_name
```
please use the proper method for getting the labels

aluminiumgeek replied:
done

akariv commented on docs/PRODUCTION-AZURE-CLUSTER.md:
````diff
@@ -0,0 +1,55 @@
+```
+
+While interactive initialization:
+- Set `enable-deis-ckan: y`
````
remove deis references.

aluminiumgeek replied:
done

aluminiumgeek committed:
- Revert using psql connection kwargs instead of string
- Rename PROVIDER_ID for azuresql prozy
- filter zk/solr pods using app name
- Remove deis reference from docs

akariv commented on ckan_cloud_operator/providers/solr/solrcloud/manager.py:
```diff
@@ -94,13 +94,13 @@ def initialize(interactive=False, dry_run=False):
+        zk_app_name = _get_resource_labels(for_deployment=True, suffix='zk')['app']
         running = len([x for x in pods['items']
-                       if x['status']['phase'] == 'Running'])
+                       if x['status']['phase'] == 'Running' and x['metadata']['labels']['app'] in [zk_app_name, sc_app_name]])
```
X['metadata']['labels']['app'].startswith('provider-solr-solrcloud')
But replace the string literal with the proper constants

aluminiumgeek replied:
done

akariv commented:
@aluminiumgeek 
One comment left, but also please make sure the tests pass.

aluminiumgeek committed:
Change check for app name

aluminiumgeek committed:
Merge branch 'master' into azure-integration

aluminiumgeek committed:
Consider zoonavigator in count

aluminiumgeek committed:
Revert moving time.sleep() to the end of loop

aluminiumgeek committed:
Restore JNLP image building: remove upper version limit for pyyaml

aluminiumgeek committed:
Freeze pyyaml version

aluminiumgeek committed:
Freezed pyyaml version for conda

akariv committed:
Merge branch 'master' into azure-integration

akariv commented on docs/DNS-AND-ROUTING.md:
```diff
@@ -0,0 +1,24 @@
+
+Code reference: `ckan_cloud_operator/routers`, `ckan_cloud_operator/providers/cluster/<cloud_provider_name>`
+
+## Adding a new DNS integration
```
Please move this bit to a doc under development/

aluminiumgeek replied:
Done

akariv commented on docs/PRODUCTION-AZURE-CLUSTER.md:
```diff
@@ -0,0 +1,53 @@
+# Creating a Production CKAN Cloud cluster using Azure Cloud Platform
```
Please follow the style in the AWS file and connect with the README (both recently modified... :) )

aluminiumgeek replied:
AWS file contains info about terraform which is not ready for Azure yet (in this branch).

I modified the doc adding prerequisites and removing not used data.

akariv reviewed (changes requested):
nearly there...

aluminiumgeek committed:
Move doc about adding a new DNS integration to development docs

aluminiumgeek committed:
Update Azure cluster docs

akariv committed:
Merge branch 'master' into azure-integration

akariv committed:
Merge branch 'master' into azure-integration

akariv committed:
Merge branch 'master' into azure-integration

aluminiumgeek committed:
Merge branch 'master' into azure-integration

aluminiumgeek committed:
Update Azure Storage module to work with updated manager

aluminiumgeek committed:
Update zk test

aluminiumgeek committed:
Add info on imagePullSecret, cronjobs values

zelima committed:
Automate CKAN instance deployment on azure k8 cluster with terraform (#105)

* Create Storage Classes with CCO.

* Encrypted creds for azure

* instal azure CLI from apt

* [azure]: Handle Azure DNS refs #112 (#115)

* Set azure as DNS provider and create DNS zone with terraform

* Docs for provisioning Ckan-Cloud with cco on azure

* [Terraform] Allos creating RG and DNS zone or passing their names

* Let traefik creating letsencrypt certificates

zelima committed:
Merge branch 'master' into azure-integration

zelima committed:
#azure add missing azure blob storage handler and fix import

zelima committed:
create storage account

zelima committed:
#azure dummy commit to trigger CI

zelima committed:
#azure retry to connect to solr

akariv commented on ckan_cloud_operator/providers/cluster/manager.py:
```diff
@@ -184,7 +184,7 @@ def create_volume(disk_size_gb, labels, use_existing_disk_name=None, zone=0):
         labels,
         **labels_manager.get_resource_labels(label_suffixes=_get_cluster_volume_label_suffixes())
     )
-    return get_provider().create_volume(disk_size_gb, labels, use_existing_disk_name=use_existing_disk_name, zone=zone)
```
why did you remove the zone parameter?

zelima replied:
Don't remember. Probably deleted while debugging and left. Bringing back

akariv commented on ckan_cloud_operator/providers/db/proxy/azuresql/manager.py:
```diff
@@ -0,0 +1,153 @@
+            'revisionHistoryLimit': 10,
+            'strategy': {'type': 'RollingUpdate', },
+            'selector': {
+                'matchLabels': _get_resource_labels(for_deployment=True, suffix=db_prefix or '')
```
use a `labels` variable instead of calling the same function 3 times

akariv commented on ckan_cloud_operator/routers/traefik/deployment.py:
```diff
@@ -89,6 +89,21 @@ def _get_deployment_spec(router_name, router_type, annotations, image=None, http
+        container['ports'].append({'containerPort': 443})
+        azure_credendials = cluster_manager.get_provider().get_azure_credentials()
+        secret_name = f'ckancloudrouter-{router_name}-azure'
+        print(azure_credendials)
```
remove

akariv commented on docs/VALUES-FILE-REFERENCE.md:
```diff
@@ -89,6 +91,7 @@
 
 - `useCloudStorage` - use a cloud storage bucket for storing CKAN files
 - `ckanPrimaryColor` - set the newly created ckan theme to be based on this color (hex based RGB, e.g. `#123456`)
+- `cronjobs` - list of CronJob entries to be created on the cluster
```
please add some details on what the value here should look like

akariv commented on docs/VALUES-FILE-REFERENCE.md:
```diff
@@ -14,6 +14,8 @@
 - `withSansSSL` - Ignored, will always be considered as True
 - `registerSubdomain` - Ignored, will always be equal to `sub-domain`
 
+- `imagePullSecret` - secret contains credentials for the private docker repo (secret is being created by CCO)
```
please add some details on what the secret should contain

zelima committed:
[cco]: post review fixes

- bring back zone for cluster providers
- bring back `return response` that was accidentally removed
- more clarity for cron values in values.yaml

zelima committed:
#azure #aws remove redundant print

zelima committed:
[#azure #aws] accept zone in create_value

zelima committed:
#azure use new version of helm for ckan

zelima committed:
handle private docker registry (#124)

zelima committed:
retry copying solr config if fails

zelima committed:
Scale down and up ckan and traefik pods so that volumes are mounted properly (#125)

* Scale down and up ckan and traefik pods so that volumes are mounted properly

* Make num replicas variable and get from values

akariv committed:
v0.1.0 #aws #azure

akariv committed:
Full paths for zkCli

akariv committed:
cont

akariv committed:
cont

akariv committed:
Log output of admin creation

akariv committed:
cont

akariv committed:
Add sleep after rescaling to settle down #aws #azure

akariv committed:
Fix terraform validation #azure

akariv committed:
Don't run providers on pullrequests #azure
